### PR TITLE
[homekit] Support lock optional characteristics on GarageDoorOpener

### PR DESCRIPTION
**Pull request by ccutrer**

### CODE REVIEW

File: bundles/org.openhab.io.homekit/README.md [hunk 1]
```diff
@@ -797,6 +797,8 @@ All accessories also support the following optional characteristic that can be l
...
+|                             |                             | LockCurrentState            | Contact, Number, String, Switch         | Current state of lock mechanism                                                                                                                                                                                                                                                                                                                               | inverted (false)                                                      | UNSECURED (0, OFF, CLOSED), SECURED (1, ON, OPEN), JAMMED (2), UNKNOWN (3)                                  |
```
Comment (lsiepel) [inline bundles/org.openhab.io.homekit/README.md:800]: If a lock is `OPEN` i would not sonsider it `SECURED`

Comment (ccutrer) [reply bundles/org.openhab.io.homekit/README.md:800]: I'm not going to argue with that, but if we want a one-off change there, it should be a separate change. This is just re-using the same characteristics as the Lock accessory.

Comment (lsiepel) [reply bundles/org.openhab.io.homekit/README.md:800]: Ok, most important is that it is consistent, and from your post i understand that it is.

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM

**Comment by grzegorz914** [comment]
@ccutrer after this update the Garage Door Opener report wrong states if door is open then report closed, if opening then report closing. Also we have no possibility to configure current and target state (enum values).

IMG_0999
IMG_0998
IMG_0997

**Comment by ccutrer** [comment]
I'm not sure how MainUI generates its configuration screen for HomeKit accessories (and I'm a little surprised that it could change automatically with any changes to the add-on), but I can assure you configuring enum values is still supported. You may need to open the Code view on the HomeKit metadata for the appropriate item. Perhaps @yfre could chime it, I believe he's done a good chunk of the work for HomeKit in MainUI?

**Comment by ccutrer** [comment]
The fix for ON/OFF not mapping correctly to to OPEN/CLOSED is #17181

**Comment by ccutrer** [comment]
https://github.com/openhab/openhab-webui/pull/2691 should fix being able to configure enum values for most/all things

**Pull request merged**